Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+                            }
+                        }
+                    }
+    
+                    if ($armResource.properties.request.apiEndPoint.contains("{{")) {
+                        # identify any placeholders in apiEndpoint
+                        $endPointUrl = $armResource.properties.request.apiEndPoint                    
+                        $placeHoldersMatched = $endPointUrl | Select-String $placeHolderPatternMatches -AllMatches
+                        
+                        if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                            # has some placeholders 
+                            $finalizedEndpointUrl = ""
+                            $finalizedEndpointUrl = "[[concat("
+                            $closureBrackets = ")]"
+    
+                            foreach ($currentPlaceHolder in $placeHoldersMatched.Matches.Value) {
+                                $placeHolderName = $currentPlaceHolder.replace("{{", "").replace("}}", "")
+                                $splitEndpoint = $endPointUrl -split "($currentPlaceHolder)"
+                                $commaCount = 0
+                                foreach($splitItem in $splitEndpoint) {
+                                    if ($splitItem -eq $currentPlaceHolder) {
+                                        if ($finalizedEndpointUrl -eq '') {
+                                            $finalizedEndpointUrl += "parameters('" + $placeHolderName + "')"
+                                        } else {
+                                            $finalizedEndpointUrl += ", parameters('" + $placeHolderName + "')"
+                                        }
+                                        
+                                        $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName "$placeHolderName" -isSecret $false
+                                    } else {
+                                        if ($commaCount -eq 0) {
+                                            $finalizedEndpointUrl += "'"+ $splitItem + "'"
+                                            $commaCount += 1
+                                        } else {
+                                            $finalizedEndpointUrl += ", '" + $splitItem + "'"
+                                        }
+                                    }
+                                }
+                            }
+    
+                            $armResource.properties.request.apiEndPoint = $finalizedEndpointUrl + $closureBrackets
+                        }
+                    }
+    
+                    # retry count placeholders
+                    if ($null -ne $armResource.properties.request.retryCount -and ($armResource.properties.request.retryCount -is [System.String] -and $armResource.properties.request.retryCount.contains("{{"))) {
```

Comment:
Check this with Prateek about the usage of retry count. It is good to have it but I am unsure to give this as deployment parameter

Reply:
removed